tasks/main.yml

```diff
@@ -1,28 +1,45 @@
 ---
+# Reboot if required before installing new updates
+- name: Check if reboot is needed
+  ansible.windows.win_powershell:
+    script: |
+      $RebootPending = Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Component Based Servicing\RebootPending"
+      $RebootRequired = $RebootPending `
+      -or (Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootRequired") `
+      -or (Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootPending")
+      if ($RebootRequired) {
+          Write-Output "YES"
```

Comment:
I suggest using the `$Ansible.Result` variable that is available using the `win_powershell` module instead of a `Write-Output`command. Also avoid using an `else` statement for these cases. Example code for this:

```
$Ansible.Result = $false
if ($RebootRequired) {
  $Ansible.Result = $true
}
```

Reply:
Code changed as suggested.